File: LogEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.WindowsAzure.Storage.Table;

namespace ConsoleTest
{
    public class LogEntity : TableEntity
    {
        public LogEntity()
        {
            PartitionKey = "a";
            RowKey = string.Format("{0:10}_{1}", DateTime.MaxValue.Ticks - DateTime.Now.Ticks, Guid.NewGuid());
        }

        public string Role { get; set; }
        public string RoleInstance { get; set; }
        public int Level { get; set; }
        public string Message { get; set; }
        public int Pid { get; set; }
        public int Tid { get; set; }
        public int EventId { get; set; }
        public DateTime EventDateTime
        {
            get
            {
                return new DateTime(long.Parse(PartitionKey.Substring(1)));
            }
        }
    }
}

File: ConsoleFSharp/LogEntity.fs
module Entities

open Microsoft.WindowsAzure.Storage.Table

type LogEntity() = 
    inherit TableEntity()
    member val Role = "" with get, set
    member val RoleInstance = "" with get, set
    member val Level = 0 with get, set
    member val Message = "" with get, set
    member val Pid = 0 with get, set
    member val Tid = 0 with get, set
    member val EventId = 0 with get, set
